org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/ProcessInfoDialog.java

```diff
@@ -216,6 +242,64 @@ public void onEvent(Event event) throws Exception {
 		if (event.getTarget() == btnOk) {
 			this.detach();
 		}
+		if(event.getTarget() == btnPrint) {
+			Clients.showBusy(Msg.getMsg(Env.getCtx(), "Processing"));
+			Clients.response(new AuEcho(ProcessInfoDialog.this, "onPrint", null));
+		}
+	}
+	
+	/**
+	 * On Print
+	 */
+	public void onPrint() {
+		// Loop through all items
+		List<File> pdfList = new ArrayList<File>();
+		for (int i = 0; i < m_logs.length; i++)
+		{
+			int recordID = m_logs[i].getRecord_ID();
+			int reportEngineType = ReportEngine.getReportEngineType(m_logs[i].getAD_Table_ID());
+			ReportEngine re = null;
+			
+			if((reportEngineType >= 0) && (recordID > 0)) {
+				re = ReportEngine.get (Env.getCtx(), reportEngineType, recordID);
+				pdfList.add(re.getPDF());
+			}
+		}
+		if (pdfList.size() > 1) {
+			try {
+				File outFile = File.createTempFile(getTitle(), ".pdf");					
+				AEnv.mergePdf(pdfList, outFile);
+
+				Clients.clearBusy();
+				Window win = new SimplePDFViewer(getTitle(), new FileInputStream(outFile));
+				SessionManager.getAppDesktop().showWindow(win, "center");
+			} catch (Exception e) {
+				log.log(Level.SEVERE, e.getLocalizedMessage(), e);

```

Comment:
Better just rethrow as runtime exception instead. Otherwise it is silent error to user.